### ISSUE TITLE
`regeneratorRuntime` is not defined

### ISSUE THREAD

**Issue by JounQin** [codemix/flow-runtime#35]
This is a:

 - Bug Report

Which concerns:

 - flow-runtime

---

### What is the current behaviour?
```
flow-runtime.min.js:3 Uncaught (in promise) ReferenceError: regeneratorRuntime is not defined
    at eval (flow-runtime.min.js:3)
    at eval (flow-runtime.min.js:3)
```
---

### What is the expected behaviour?

Throw error

Please add `babel-plugin-transform-runtime` is your `.babelrc`

---

### Which package versions are you using?

0.2.1

**Comment by phpnode** [comment]
(I need to add some docs for this):

I don't want to bundle `regenerator-runtime` with the library - it's a largish dependency and there's a good chance that the user is already including code which depends on it. If we bundle it they'll get a duplicate copy and that's bad.

The answer is to do the following somewhere before you import flow-runtime:

```js
import 'regenerator-runtime/runtime';
import t from 'flow-runtime';
```

Ordinarily I'd add `regenerator-runtime` to `peerDependencies`, but I'm hesitant to as it's only used by the minified build - supported node versions and modern browsers have real generators.

**Comment by JounQin** [comment]
I use `flow-runtime` just for development with `babel-plugin-flow-runtime`, is that a good idea to require `regenerator-runtime` for us?

**Comment by phpnode** [comment]
@JounQin a couple of solutions, you could do a conditional require if you're in development mode (and dead code elimination tools will get rid of it):

```js
if (process.env.NODE_ENV === 'development') {
  require('regenerator-runtime/runtime');
}
```

or you could remap the module in your webpack configuration to use `flow-runtime.js` rather than `flow-runtime.min.js`. If you're only testing with modern browsers this is probably the best idea.

**Comment by reaperes** [comment]
In my case, I add a line of code in webpack.dev.config file. Please comments if it's not correct.

```
entry: {
  app: [
    'regenerator-runtime/runtime',
    'webpack-hot-middleware/client',
    './app/index.js'
  ]
},
```

**Comment by aretecode** [comment]
I also have this issue 
![image](https://cloud.githubusercontent.com/assets/4022631/22275043/b552abf0-e25e-11e6-9f90-1b53d38c16e2.png)

even though regenerator is in my babel plugins:
```js
{ 
plugins:
   [ 'transform-runtime',
     'transform-regenerator',
     'transform-async-to-generator',
     [ 'flow-runtime', [Object] ],
     'transform-flow-strip-types',
     'transform-class-properties',
     'transform-object-rest-spread',
     'transform-decorators-legacy',
     'add-module-exports',
     'transform-react-jsx' ],
  presets: [ 'es2015', 'stage-0' ],
  babelrc: false,
  cacheDirectory: true 
}
```

even if I import it in my entry point:
![image](https://cloud.githubusercontent.com/assets/4022631/22275072/f33613c6-e25e-11e6-9322-6572ed48993b.png)

**Comment by phpnode** [comment]
@aretecode regenerator-runtime must be included separately, it's not a babel plugin. If you're importing it in your entry point are you doing so *before* flow-runtime itself is required/imported? 

also I see a `Not Implemented` error in your screenshot - please can you post some code which replicates that?

**Comment by SaurabhKshirsagar** [comment]
babel-polyfill is required for regenerator-runtime. You must install it in order.
`npm i -D babel-core babel-polyfill babel-preset-es2015 babel-preset-stage-0 babel-loader`

.babelrc
```
{
    "presets": ["stage-2", "react","es2015"],
    "plugins": [["flow-runtime"], "transform-decorators-legacy"]
}
```

In your app startup file require _babel-core/register and babel-polyfill_ 
```
require("babel-core/register");
require("babel-polyfill");
```

If you are using webpack you need to put it as the first entry as _babel-polyfill_
```
module.exports = {
  entry: ['babel-polyfill', './test_index.js']
}
```

**Comment by vjpr** [comment]
Adding `babel-regenerator-runtime` to the top of my webpack main bundle worked for me. Everything else did not.

**Comment by ForbesLindesay** [comment]
The correct solution to this is to add https://babeljs.io/docs/plugins/transform-runtime/ to the babel config for flow-runtime.  You do not need to worry about this causing a duplicate version of the regenerator runtime to exist, because flow-runtime should not be used in production anyway, and the regenerator-runtime added by babel-plugin-transform-runtime does not pollute the global scope, so it won't conflict with any applications that use regenerator themselves.  Without this, libraries that use flow-runtime will not work out of the box.

I'm going to be forced to abandon flow-runtime until this is fixed.

**Comment by phpnode** [comment]
Ok, I agree this is not worth the pain as most people are not using it in prod, will look into this today.

**Comment by ForbesLindesay** [comment]
I've submitted a pull request that I believe should fix this issue.

**Comment by phpnode** [comment]
released in 0.11.1
